rfc_process.md

```diff
@@ -0,0 +1,83 @@
+# Versioning Process for the OpenTracing Specification
+The OpenTracing specification consists of a cross-language, abstract interface, along with an attendant set of language-specific interfaces. Every interface is versioned, with each language interface supporting a version of the abstract interface.
+
+In order effect changes to the OpenTracing specification, all interfaces are developed together in a three step process:
+
+* DRAFT: An RFC is drafted to define and justify the desired changes.
+* TEST: A new version of each langauage interface is designed, tested, and released.
+* ACCEPTED: The abstract interface is versioned to reflect the final changes.
+
+# Proposal Components
+
+## RFC (Request for Change)
+Proposals begin as a single document, written in markdown and committed to the `/rfc` directory.
+
+An RFC consists of the following:
+
+* Status, date, and author.
+* Problem statement.
+* Historical background.
+* Abstract interface.
+* Use cases.
+* Risk Assesment.
+
+An RFC has the following qualities:
+
+* The issue is clearly defined, and within the scope of the OpenTracing charter.
+* The utility to the OT community is illustrated with concrete use cases. It should be clear not only what will change, but how the new interface is expected to be used.
+* Changes to the abstract interface are clearly defined. Language-specific interfaces will be implemented in a later phase,but the abstract interface must be clear enough that the changes could be ported to many languages without much deviation.
+* All expected behavior should be documented, in sufficient detail such that language-specific tests and examples can be generated from the descriptions.
+* Risks, backwards compatibility, and potential for inconsistency are addressed.
+
+## Tracking Issue
+Often, a proposal will consist of multiple artifacts, such as GitHub issues, pull requests, release candidates, etc. In order to manage these resources, a single, long-running tracking issue is assigned to each proposal.
+
+A tracking issue consists of the following:
+
+* Status, date, and author.
+* Short description of the proposal.
+* A link to the committed RFC.
+* Lists of open issues, PRs, and upcoming deadlines.
+* Links to relevant artifacts, such as release candidates.
+* Links to meetings and other discussion channels.
+
+## Release Candidates
+Each language interface implements the abstract interface contained in the RFC.
+
+A complete release candidate consists of the following:
+* Repository branch containing the new version.
+* Packages for installing the new version.
+* Tracers which bind to the new version.
```

Comment:
How should we define/choose which tracer should try to implement a feature? Do we just rely on the fact that people here are working on tracers anyway and will implement features in "their" tracers? Also, tracers are more or less separate projects with separate release cyclesoutside of our direct control (but tied to us of course). Maybe it would be easier if every language had the "basic tracer" which could be used to show the usage of the feature?

Reply:
In general, I believe we should use "rough consensus" for making these decisions. No one can use an API without at least one tracer and one complete set of instrumentation (framework plus RPC library). So an API can't really be examined without these. Ideally, a new release candidate is not made official without at least support from some tracers and instrumentation, or it is useless to people. As to which ones and how many, that should be a case by case basis. For example, which tracers support C# might be different from those which support C++.

We definitely need to revisit the Basic Tracers and make them more useful, but I don't think they are a substitute.